lib/Extension/CodeTransformExtra/Rpc/GenerateAccessorHandler.php

```diff
@@ -33,27 +46,86 @@ public function name(): string
 
     public function configure(Resolver $resolver)
     {
+        $resolver->setDefaults([
+            self::PARAM_OFFSET => null,
+            self::PARAM_NAMES => null,
+        ]);
         $resolver->setRequired([
             self::PARAM_PATH,
             self::PARAM_SOURCE,
-            self::PARAM_OFFSET,
         ]);
     }
 
     public function handle(array $arguments)
     {
-        $sourceCode = $this->generateAccessor->generateAccessor(
-            SourceCode::fromStringAndPath(
-                $arguments[self::PARAM_SOURCE],
-                $arguments[self::PARAM_PATH]
-            ),
-            $arguments[self::PARAM_OFFSET]
-        );
+        $class = $this->class($arguments[self::PARAM_SOURCE]);
+
+        if (null === $arguments[self::PARAM_OFFSET]) {
+            $this->requireInput(ListInput::fromNameLabelChoices(
+                self::PARAM_NAMES,
+                sprintf('Properties from "%s"', $class->name()),
+                $this->propertiesChoices($class)
+            )->withMultiple(true));
+        }
+
+        if ($this->hasMissingArguments($arguments)) {
+            return $this->createInputCallback($arguments);
+        }
+
+        $originalSource = $arguments[self::PARAM_SOURCE];
+        $newSource = SourceCode::fromStringAndPath($originalSource, $arguments[self::PARAM_PATH]);
+
+        if ($offset = $arguments[self::PARAM_OFFSET]) {
+            $newSource = $this->generateAccessor->generateFromOffset($newSource, $offset);
+        } else {
```

Comment:
Note: As a general rule `else` is treated as a code smell in this project, so would always prefer to delegate to a method with an early return. (but I guess we remove the offset anyway).